transcription/ibm.go

```diff
@@ -0,0 +1,134 @@
+package transcription
+
+import (
+	"bufio"
+	"encoding/base64"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// IBMResult is the result of an IBM transcription. See
+// https://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/doc/speech-to-text/output.shtml
+// for details.
+type IBMResult struct {
+	ResultIndex int              `json:"result_index"`
+	Results     []ibmResultField `json:"results"`
+}
+type ibmResultField struct {
+	Alternatives []ibmAlternativesField `json:"alternatives"`
+	Final        bool                   `json:"final"`
+}
+type ibmAlternativesField struct {
+	WordConfidence    []ibmWordConfidence `json:"word_confidence"`
+	OverallConfidence float64             `json:"confidence"`
+	Transcript        string              `json:"transcript"`
+	Timestamps        []ibmWordTimestamp  `json:"timestamps"`
+}
+type ibmWordConfidence []interface{}
+type ibmWordTimestamp []interface{}
```

Comment:
Yes, good question. 

From ["How to use interfaces in Go"](http://jordanorelli.com/post/32665860244/how-to-use-interfaces-in-go)

> The interface{} type, the empty interface is the interface that has no methods.

> Since there is no implements keyword, all types implement at least zero methods, and satisfying an interface is done automatically, all types satisfy the empty interface.
That means that if you write a function that takes an interface{} value as a parameter, you can supply that function with any value.

You will notice that the arrays returned by IBM contain both a string and a float64. Since typed slices in Go must contain values of the **same** type, I created a slice of type interface{} since both strings and float64 satisfy the empty interface.